core/src/main/java/org/apache/sdap/mudrod/driver/ESDriver.java

```diff
@@ -473,12 +474,20 @@ protected Client makeClient(Properties props) throws IOException {
 
     Client client = null;
 
-    // Prefer TransportClient
+    // Prefer to TransportClient
     if (hosts != null && port > 1) {
-      TransportClient transportClient = new ESTransportClient(settings);
-      for (String host : hosts)
-        transportClient.addTransportAddress(new InetSocketTransportAddress(InetAddress.getByName(host), port));
-      client = transportClient;
+      TransportClient transportClient = null;
+      try {
+        transportClient = new ESTransportClient(settings);
+        for (String host : hosts){
+          transportClient.addTransportAddress(new InetSocketTransportAddress(InetAddress.getByName(host), port));
+        }
+        client = transportClient;
+      } catch (Exception e) {
+        
+      } finally {
```

Comment:
You've added the finally statement, but you have not closed the client. Just use the try-with-resources statement OK.